python/packages/autogen-core/src/autogen_core/_pydantic_compat.py

```diff
@@ -23,9 +23,9 @@ def evaluate_forwardref(
 
         return evaluate_forwardref_internal(value, globalns, localns)
     else:
-        from pydantic._internal._typing_extra import eval_type_lenient
+        from pydantic._internal._typing_extra import try_eval_type
 
-        return eval_type_lenient(value, globalns, localns)
+        return try_eval_type(value, globalns, localns)[0]
```

Comment:
I didn't realize we were depending on internal things here. I wonder if this can be rewritten to not depend on Pydantic internals

Reply:
I've opened a PR for removing this internal pydantic use, so it should make this PR easier #4816

Reply:
#4816 is merged. @srjoglekar246

Reply:
evaluate_forwardref is no longer need, so in resolving the conflict we can accept changes from main.